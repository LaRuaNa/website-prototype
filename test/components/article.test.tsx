import React from 'react';
import Article from '../../src/components/article';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import {
  createLearnPageData,
  createLearnPageContext,
  createLearnPageTocData,
} from '../__fixtures__/page';

describe('Article component', () => {
  it('renders correctly', () => {
    const renderer = ShallowRenderer.createRenderer();
    const learnPageData = createLearnPageData();
    const learnPageContext = createLearnPageContext();
    const learnPageTocData = createLearnPageTocData();

    const {
      doc: {
        frontmatter: { title, description },
        html,
        fields: { authors },
      },
    } = learnPageData;

    const { relativePath, next, previous } = learnPageContext;

    const { learnPageToc } = learnPageTocData;

    renderer.render(
      <Article
        title={title}
        html={html}
        next={next}
        previous={previous}
        authors={authors}
        relativePath={relativePath}
        tableOfContents={learnPageToc}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
