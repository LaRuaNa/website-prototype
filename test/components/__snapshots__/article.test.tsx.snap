// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders correctly 1`] = `
<article
  className="article-reader"
>
  <h1
    className="article-reader__headline"
  >
    test-title
  </h1>
  <TOC
    heading="TABLE OF CONTENTS"
    tableOfContents="<ul>
<li><a href=\\"/introduction-to-nodejs/#a-vast-number-of-libraries\\">A Vast Number of Libraries</a></li>
<li><a href=\\"/introduction-to-nodejs/#an-example-nodejs-application\\">An Example Node.js Application</a></li>
<li><a href=\\"/introduction-to-nodejs/#nodejs-frameworks-and-tools\\">Node.js Frameworks and Tools</a></li>
</ul>"
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<span>Test html</span>",
      }
    }
  />
  <AuthorsList
    authors={
      Array [
        "test-user1",
        "test-user2",
      ]
    }
  />
  <EditLink
    relativePath="test-path"
  />
  <Pagination
    next={
      Object {
        "slug": "test-slug",
        "title": "test-title",
      }
    }
    previous={
      Object {
        "slug": "test-slug",
        "title": "test-title",
      }
    }
  />
</article>
`;
